depthai_ros_driver/cmake/ros_version_agnostic_add_node.cmake

```diff
@@ -0,0 +1,141 @@
+    message(FATAL_ERROR
+      "add_ros_node() called with unused arguments: '"
+      "${_ARG_UNPARSED_ARGUMENTS}'")
+  endif()
+
+  if($ENV{ROS_VERSION} STREQUAL 1)
+    message(STATUS "Compiling ${name} as ROS 1")
+
+    if(NOT ${SKIP_PACKAGE_SEARCH})
+      find_package(catkin REQUIRED COMPONENTS ${ARGN})
+    else()
+      message(DEBUG "Skipping package search")
+    endif()
+
+    catkin_package (
+      INCLUDE_DIRS ${_ARG_INCLUDE_DIRS}
+        LIBRARIES ${name}_nodelet
+        CATKIN_DEPENDS ${_ARG_PKG_DEPS}
+        DEPENDS ${_ARG_SYSTEM_DEPS}
+    )
+
+    include_directories(${_ARG_INCLUDE_DIRS})
+    include_directories(SYSTEM ${catkin_INCLUDE_DIRS})
+    foreach(sys_dep ${_ARG_SYSTEM_DEPS})
+      include_directories(SYSTEM ${${sys_dep}_INCLUDE_DIRS})
+    endforeach()
+
+    add_library(${name}_lib ${_ARG_SOURCES} ${_ARG_ROS1_SOURCES})
+    target_link_libraries(${name}_lib ${catkin_LIBRARIES})
+    foreach(sys_dep ${_ARG_SYSTEM_DEPS})
+      target_link_libraries(${name}_lib ${${sys_dep}_LIBRARIES})
+    endforeach()
+    foreach(target_dep ${_ARG_TARGET_DEPS})
+      target_link_libraries(${name}_lib ${target_dep})
+    endforeach()
+
+    add_dependencies(${name}_lib
+      ${${name}_EXPORTED_TARGETS} ${catkin_EXPORTED_TARGETS})
+
+    add_library(${name}_nodelet ${_ARG_ROS1_COMPONENT_SOURCES})
+    target_link_libraries(${name}_nodelet ${name}_lib)
+
+    add_executable(${name} ${_ARG_ROS1_EXE_SOURCES})
+    target_link_libraries(${name} ${name}_lib)
+
+    install(TARGETS ${name}_lib ${name}_nodelet ${name}
+      ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+      LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+      RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION})
+
+    install(DIRECTORY ${_ARG_DIRECTORIES}
+      DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
+      USE_SOURCE_PERMISSIONS
+    )
+
+    set(${name}_PACKAGE_BIN_DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION})
+    set(${name}_PACKAGE_LIB_DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION})
+    set(${name}_PACKAGE_SHARE_DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION})
+  else()
+    message(STATUS "Compiling ${name} as ROS 2")
+
+    if(NOT ${SKIP_PACKAGE_SEARCH})
+      find_package(ament_cmake REQUIRED)
+      foreach(pkg ${ARGN})
+        find_package(${pkg} REQUIRED)
+      endforeach()
+    else()
+      message(DEBUG "Skipping package search")
+    endif()
+
+    add_definitions(-DUSE_ROS2)
+
+    include_directories(${_ARG_INCLUDE_DIRS})
+
+    add_library(${name}_lib SHARED ${_ARG_SOURCES} ${_ARG_ROS2_SOURCES})
+    foreach(sys_dep ${_ARG_SYSTEM_DEPS})
+      target_link_libraries(${name}_lib PUBLIC ${${sys_dep}_LIBRARIES})
+    endforeach()
+    foreach(target_dep ${_ARG_TARGET_DEPS})
+      target_link_libraries(${name}_lib PUBLIC ${target_dep})
+    endforeach()
+    ament_target_dependencies(${name}_lib PUBLIC ${_ARG_PKG_DEPS})
+
+    rclcpp_components_register_nodes(${name}_lib ${_ARG_COMPONENT_CLASS})
```

Comment:
should be `${_ARG_ROS2_COMPONENT_CLASS}`

Reply:
Fixed in 5d03b43

Reply:
Fixed in 4a36aa0